test: make test-fs-watchfile reliable

Trott opened the pull request:
Fixes: https://github.com/nodejs/node/issues/13377

<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)
<!-- Provide affected core subsystem(s) (like doc, cluster, crypto, etc). -->
test fs

Trott commented:
Relevant stress tests:

AIX: https://ci.nodejs.org/job/node-stress-single-test/1263/nodes=aix61-ppc64/console
macOS: https://ci.nodejs.org/job/node-stress-single-test/nodes=osx1010/1265/console

Trott commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/8428/

Trott commented:
Stress tests indicate this doesn't fix the flakiness on AIX.  I'm starting to wonder if sometimes AIX hands us watchers that will simply never fire. The `fs.watch()` stuff is highly specific to the OS.

gireeshpunathil commented:
[Background](https://github.com/nodejs/node/blob/master/test/parallel/test-fs-watch-encoding.js#L21)

Watch facility on folders are not fool-proof in AIX, and we have skipped those tests which do that. I guess the new changes in 
[test-fs-watchfile.js](https://github.com/nodejs/node/blob/master/test/parallel/test-fs-watchfile.js#L68) is not catering to that.

[#13111](https://github.com/nodejs/node/pull/13111) made that change,  it's original intent was to make sure the filename argument appears in the callback, when it fires - so I requested to include AIX as well, as the filename argument is availabe in that platform. But then the change introduced folder watch and hence the flaky result.

Proposals - one of:
1. Amend the new change in test-fs-watchfile.js to skip AIX.
2. Amend the new change in test-fs-watchfile.js to file watch as opposed to folder watch.

Trott commented:
Thanks, @gireeshpunathil! I've moved the test back to parallel and updated it to skip the directory watch test on AIX.

CI: https://ci.nodejs.org/job/node-test-pull-request/8440/

Trott commented:
(Whoops, will need to update the commit message when landing.)

Trott commented:
@refack I've change the error throwing stuff back to the way it was. I don't feel strongly about it and I have mixed feelings about doing small-but-unrelated refactors with more meaningful changes anyway. So happy to put it back the way you like. It also gave me an opportunity to fix the commit message!

CI again: https://ci.nodejs.org/job/node-test-pull-request/8442/

Trott commented:
Test failure on OS X is unrelated (and is something that will be fixed coincidentally by a NodeTodo task I just gave to someone yesterday).

Thoughts on merging this now rather than waiting the 72 hours on the grounds that it's trivial and fixes a CI issue?

refack commented:
> Thoughts on merging this now rather than waiting the 72 hours on the grounds that it's trivial and fixes a CI issue?

👍

Trott commented:
Landed in 98aa25c765f50eab80f27750b9b7f82888af4c8c
